From 49ee1f8153e15aab92a8a5732bdb05856204f01d Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 12 Apr 2016 12:21:09 +1000
Subject: [PATCH 0556/1722] drm/amd/dal: constify opp/ipp function pointers.

This also fixes having funcs in the global namespace due to missing
static on one set.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c   | 2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c   | 2 +-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_v.c | 9 +--------
 drivers/gpu/drm/amd/dal/dc/dce80/dce80_ipp.c     | 2 +-
 drivers/gpu/drm/amd/dal/dc/dce80/dce80_opp.c     | 2 +-
 drivers/gpu/drm/amd/dal/dc/inc/hw/ipp.h          | 2 +-
 drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h          | 2 +-
 7 files changed, 7 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
index 3a9756d..dd69f60 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_ipp.c
@@ -31,7 +31,7 @@
 
 #include "dce110_ipp.h"
 
-static struct ipp_funcs funcs = {
+static const struct ipp_funcs funcs = {
 		.ipp_cursor_set_attributes = dce110_ipp_cursor_set_attributes,
 		.ipp_cursor_set_position = dce110_ipp_cursor_set_position,
 		.ipp_program_prescale = dce110_ipp_program_prescale,
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
index 6b9bf2d..698ec2f 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp.c
@@ -42,7 +42,7 @@ enum {
 /* Constructor, Destructor               */
 /*****************************************/
 
-static struct opp_funcs funcs = {
+static const struct opp_funcs funcs = {
 		.opp_power_on_regamma_lut = dce110_opp_power_on_regamma_lut,
 		.opp_set_csc_adjustment = dce110_opp_set_csc_adjustment,
 		.opp_set_csc_default = dce110_opp_set_csc_default,
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_v.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_v.c
index 2ca5b77..64baff6 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_v.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_opp_v.c
@@ -38,21 +38,14 @@
 /* Constructor, Destructor               */
 /*****************************************/
 
-struct opp_funcs funcs = {
+static const struct opp_funcs funcs = {
 		.opp_power_on_regamma_lut = dce110_opp_power_on_regamma_lut_v,
-
 		.opp_program_regamma_pwl = dce110_opp_program_regamma_pwl_v,
-
 		.opp_set_csc_default = dce110_opp_v_set_csc_default,
-
 		.opp_set_csc_adjustment = dce110_opp_v_set_csc_adjustment,
-
 		.opp_set_dyn_expansion = dce110_opp_set_dyn_expansion,
-
 		.opp_set_regamma_mode = dce110_opp_set_regamma_mode,
-
 		.opp_destroy = dce110_opp_destroy,
-
 		.opp_program_fmt = dce110_opp_program_fmt,
 };
 
diff --git a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_ipp.c b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_ipp.c
index 2ddcffd..86826c2 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_ipp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_ipp.c
@@ -33,7 +33,7 @@
 
 #include "dce110/dce110_ipp.h"
 
-static struct ipp_funcs funcs = {
+static const struct ipp_funcs funcs = {
 		.ipp_cursor_set_attributes = dce110_ipp_cursor_set_attributes,
 		.ipp_cursor_set_position = dce110_ipp_cursor_set_position,
 		.ipp_program_prescale = dce110_ipp_program_prescale,
diff --git a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_opp.c b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_opp.c
index 3c2358a..b69e8a5 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce80/dce80_opp.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce80/dce80_opp.c
@@ -76,7 +76,7 @@ static const struct dce80_opp_reg_offsets reg_offsets[] = {
 }
 };
 
-static struct opp_funcs funcs = {
+static const struct opp_funcs funcs = {
 		.opp_power_on_regamma_lut = dce80_opp_power_on_regamma_lut,
 		.opp_set_csc_adjustment = dce80_opp_set_csc_adjustment,
 		.opp_set_csc_default = dce80_opp_set_csc_default,
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/hw/ipp.h b/drivers/gpu/drm/amd/dal/dc/inc/hw/ipp.h
index c448346..ceeca07 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/hw/ipp.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/hw/ipp.h
@@ -36,7 +36,7 @@
 struct input_pixel_processor {
 	struct  dc_context *ctx;
 	uint32_t inst;
-	struct ipp_funcs *funcs;
+	const struct ipp_funcs *funcs;
 };
 
 enum ipp_prescale_mode {
diff --git a/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h b/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
index b7dc40b9..ebe966c 100644
--- a/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
+++ b/drivers/gpu/drm/amd/dal/dc/inc/hw/opp.h
@@ -204,7 +204,7 @@ enum opp_regamma {
 struct output_pixel_processor {
 	struct dc_context *ctx;
 	uint32_t inst;
-	struct opp_funcs *funcs;
+	const struct opp_funcs *funcs;
 };
 
 enum fmt_stereo_action {
-- 
2.7.4

